### PR TITLE
[aci] amend the stables IR curve - August edition

### PR DESCRIPTION
**Pull request by Rozengarden**

### CODE REVIEW

File: src/20240829_Multi_StablecoinIRCurveAmendment/StablecoinIRCurveAmendment_20240829.s.sol [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    GovV3Helpers.createPayload(actions);
+  }
+}
+
+/**
+ * @dev Deploy Avalanche
+ * deploy-command: make deploy-ledger contract=src/20240829_Multi_StablecoinIRCurveAmendment/StablecoinIRCurveAmendment_20240829.s.sol:DeployAvalanche chain=avalanche
+ * verify-command: FOUNDRY_PROFILE=avalanche npx catapulta-verify -b broadcast/StablecoinIRCurveAmendment_20240829.s.sol/43114/run-latest.json
+ */
+contract DeployAvalanche is AvalancheScript {
```
Comment (brotherlymite) [inline src/20240829_Multi_StablecoinIRCurveAmendment/StablecoinIRCurveAmendment_20240829.s.sol:72]: Unused should be removed

File: src/20240829_Multi_StablecoinIRCurveAmendment/StablecoinIRCurveAmendment_20240829.s.sol [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actionsPolygon = new IPayloadsControllerCore.ExecutionAction[](1);
+    actionsPolygon[0] = GovV3Helpers.buildAction(
+      type(AaveV3Polygon_StablecoinIRCurveAmendment_20240829).creationCode
+    );
+    payloads[1] = GovV3Helpers.buildPolygonPayload(vm, actionsPolygon);
+
+    IPayloadsControllerCore.ExecutionAction[]
+      memory actionsAvalanche = new IPayloadsControllerCore.ExecutionAction[](2);
+    actionsAvalanche[0] = GovV3Helpers.buildAction(
```
Comment (brotherlymite) [inline src/20240829_Multi_StablecoinIRCurveAmendment/StablecoinIRCurveAmendment_20240829.s.sol:254]: Unused remove and adjust the script accordingly

File: diffs/AaveV2Avalanche_StablecoinIRCurveAmendment_20240829_before_AaveV2Avalanche_StablecoinIRCurveAmendment_20240829_after.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+## Reserve changes
```
Comment (brotherlymite) [inline diffs/AaveV2Avalanche_StablecoinIRCurveAmendment_20240829_before_AaveV2Avalanche_StablecoinIRCurveAmendment_20240829_after.md:1]: Unused please remove

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV2Ethereum_StablecoinIRCurveAmendment_20240829.sol [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    returns (IAaveV2ConfigEngine.RateStrategyUpdate[] memory)
+  {
+    IAaveV2ConfigEngine.RateStrategyUpdate[]
+      memory rateStrategies = new IAaveV2ConfigEngine.RateStrategyUpdate[](3);
+    rateStrategies[0] = IAaveV2ConfigEngine.RateStrategyUpdate({
+      asset: AaveV2EthereumAssets.USDT_UNDERLYING,
+      params: IV2RateStrategyFactory.RateStrategyParams({
+        optimalUtilizationRate: EngineFlags.KEEP_CURRENT,
+        baseVariableBorrowRate: EngineFlags.KEEP_CURRENT,
+        variableRateSlope1: _bpsToRay(5_50),
```
Comment (brotherlymite) [inline src/20240829_Multi_StablecoinIRCurveAmendment/AaveV2Ethereum_StablecoinIRCurveAmendment_20240829.sol:29]: @Rozengarden any reason to reduce rates on Aave V2 Ethereum and not on Polygon and Avax V2 Markets?

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Arbitrum_StablecoinIRCurveAmendment_20240829.sol [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    rateStrategies[5] = IAaveV3ConfigEngine.RateStrategyUpdate({
+      asset: AaveV3ArbitrumAssets.FRAX_UNDERLYING,
+      params: IAaveV3ConfigEngine.InterestRateInputData({
+        optimalUsageRatio: EngineFlags.KEEP_CURRENT,
+        baseVariableBorrowRate: EngineFlags.KEEP_CURRENT,
+        variableRateSlope1: 5_50,
+        variableRateSlope2: EngineFlags.KEEP_CURRENT
+      })
+    });
+
```
Comment (brotherlymite) [inline src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Arbitrum_StablecoinIRCurveAmendment_20240829.sol:77]: @Rozengarden does it makes sense to reduce the rates for GHO on arbitrum as well?

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Ethereum_StablecoinIRCurveAmendment_20240829.sol [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    rateStrategies[6] = IAaveV3ConfigEngine.RateStrategyUpdate({
+      asset: AaveV3EthereumAssets.PYUSD_UNDERLYING,
+      params: IAaveV3ConfigEngine.InterestRateInputData({
+        optimalUsageRatio: EngineFlags.KEEP_CURRENT,
+        baseVariableBorrowRate: EngineFlags.KEEP_CURRENT,
+        variableRateSlope1: 5_50,
+        variableRateSlope2: EngineFlags.KEEP_CURRENT
+      })
+    });
+
```
Comment (brotherlymite) [inline src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Ethereum_StablecoinIRCurveAmendment_20240829.sol:86]: @Rozengarden does it make sense to also include USDe

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Polygon_StablecoinIRCurveAmendment_20240829.sol [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      asset: AaveV3PolygonAssets.jEUR_UNDERLYING,
+      params: IAaveV3ConfigEngine.InterestRateInputData({
+        optimalUsageRatio: EngineFlags.KEEP_CURRENT,
+        baseVariableBorrowRate: EngineFlags.KEEP_CURRENT,
+        variableRateSlope1: 5_50,
+        variableRateSlope2: EngineFlags.KEEP_CURRENT
+      })
+    });
+    rateStrategies[4] = IAaveV3ConfigEngine.RateStrategyUpdate({
+      asset: AaveV3PolygonAssets.EURA_UNDERLYING,
```
Comment (brotherlymite) [inline src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Polygon_StablecoinIRCurveAmendment_20240829.sol:60]: as we are including frozen assets, might as well include miMATIC? no strong opinion on it as it has only $3.5k borrowed

File: src/20240829_Multi_StablecoinIRCurveAmendment/StablecoinIRCurveAmendment_20240829.s.sol (hunk 1 above)
Comment (brotherlymite) [reply src/20240829_Multi_StablecoinIRCurveAmendment/StablecoinIRCurveAmendment_20240829.s.sol:72]: This is breaking the compilation as well

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV2Ethereum_StablecoinIRCurveAmendment_20240829.sol (hunk 4 above)
Comment (Rozengarden) [reply src/20240829_Multi_StablecoinIRCurveAmendment/AaveV2Ethereum_StablecoinIRCurveAmendment_20240829.sol:29]: From CL: mainnet tvl is bigger and the polygon and avalanche market were addressed separately: https://app.aave.com/governance/v3/proposal/?proposalId=154

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Arbitrum_StablecoinIRCurveAmendment_20240829.sol (hunk 5 above)
Comment (Rozengarden) [reply src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Arbitrum_StablecoinIRCurveAmendment_20240829.sol:77]: from CL: GHO should stay 1% point above mainnet rate

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Polygon_StablecoinIRCurveAmendment_20240829.sol (hunk 7 above)
Comment (Rozengarden) [reply src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Polygon_StablecoinIRCurveAmendment_20240829.sol:60]: from CL: no

File: src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Ethereum_StablecoinIRCurveAmendment_20240829.sol (hunk 6 above)
Comment (Rozengarden) [reply src/20240829_Multi_StablecoinIRCurveAmendment/AaveV3Ethereum_StablecoinIRCurveAmendment_20240829.sol:86]: from CL: USDe should not be included given that it has been occasionally over-utilized, creating rate volatility

**Review by brotherlymite — APPROVED** [review]

**Review by pavelvm5 — APPROVED** [review]

**Pull request merged**